QMLComponents/analysisform.cpp

```diff
@@ -308,12 +307,31 @@ QString AnalysisForm::msgsListToString(const QStringList & list) const
 	return !text.size() ? "" : "<ul style=\"margins:0px\">" + text + "</ul>";
 }
 
-void AnalysisForm::setInfo(QString info)
+void AnalysisForm::setInfo(const QVariant& info)
 {
-	if (_info == info)
+	if (_info.var == info)
 		return;
 
-	_info = info;
+	_info.var = info;
+
+	if (info.canConvert<QString>())
+		_info.top = info.toString();
+	else if (info.canConvert<QMap<QString, QVariant> >())
+	{
+		QMap<QString, QVariant> map = info.toMap();
```

Comment:
Maybe you should make a proper type for this instead of a variantmap...

Reply:
Hmm I have tried now for some hours, but I cannot make it.
If I have a look at the Qt doc, I should use Q_GADGET and QML_VALUE_TYPE
https://doc.qt.io/qt-6/qtqml-typesystem-valuetypes.html
https://doc.qt.io/qt-6/qtqml-cppintegration-definetypes.html
But it gives errors either in compilation at run-time not recognizing the type... There is also no real example how to use QML_VALUE_TYPE.

Reply:
https://doc.qt.io/qt-6/qtqml-cppintegration-definetypes.html#registering-an-instantiable-object-type

This no?

Reply:
I can have a look tomorrow